prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,341 @@
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.annotation.JsonProperty
+
+import java.time.ZonedDateTime
+import java.time.Duration
+import java.time.format.DateTimeFormatter
+import com.cronutils.model.time.ExecutionTime;
+import com.cronutils.parser.CronParser;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.CronType;
+
+import com.sendgrid.helpers.mail.objects.*;
+import com.sendgrid.*;
+import com.sendgrid.helpers.mail.Mail;
+
+import gov.cdc.prime.router.azure.db.enums.SettingType;
+import org.jooq.Configuration
+
+import java.io.IOException;
+
+import khttp.get as httpGet
+import org.json.JSONObject
+import org.json.JSONArray
+
+import gov.cdc.prime.router.secrets.SecretHelper;
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.HttpStatus
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.BindingName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+
+import com.okta.jwt.JwtVerifiers
+import java.util.logging.Level
+import java.util.logging.Logger
+
+const val SCHEDULE = "*/5 * * * *" // every 5 minutes
+const val OKTA_ISSUER = "https://hhs-prime.okta.com/oauth2/default"
+const val OKTA_GROUPS_API = "https://hhs-prime-admin.okta.com/api/v1/groups"
+const val FROM_EMAIL = "reportstream@cdc.gov"
+const val SUBJECT_EMAIL = "ReportStream Daily Email"
+
+
+
+data class EmailSchedule ( 
+    val template: String,
+    val type: String,
+    val cronSchedule: String,
+    val organizations: List<String> = ArrayList<String>(),
+    val emails: List<String> = ArrayList<String>(),
+    val parameters: Map<String,String> = HashMap<String,String>()
+) {}
+ 
+class EmailScheduleEngine  {
+
+    val workflowEngine = WorkflowEngine()
+
+    /**
+     * Create Email Schedule
+     */
+    @FunctionName("createEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun createEmailSchedule(
+        @HttpTrigger(
+            name = "createEmailSchedule",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context.logger );
+        var ret = request.createResponseBuilder(HttpStatus.UNAUTHORIZED);
+
+        if( user !== null ){
+            val id = WorkflowEngine.databaseAccess.insertEmailSchedule(request.body, user );
+            ret.status(HttpStatus.CREATED)
+            ret.body( "created id $id")
```

Comment:
For automation purposes, I think it would make more sense to just return the id without any additional strings (i.e. `ret.body( "$id")`). I'm think about people wanting to do something with the result in a script that uses `curl`.

Reply:
Agree - changing